### PR TITLE
Optimize Snapshot

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Concurrently working on snapshot and control the go routines number

### CODE REVIEW

File: pkg/scheduler/api/node_info.go [hunk 1]
```diff
@@ -229,8 +229,7 @@ func (ni *NodeInfo) AddTask(task *TaskInfo) error {
 
 	key := PodKey(task.Pod)
 	if _, found := ni.Tasks[key]; found {
-		return fmt.Errorf("task <%v/%v> already on node <%v>",
-			task.Namespace, task.Name, ni.Name)
+		return nil
```
Comment (k82cn) [inline pkg/scheduler/api/node_info.go:232]: why return nil?

Comment (hzxuzhonghu) [reply pkg/scheduler/api/node_info.go:232]: I am not sure the logic before, it looks this happens when calling AddTask multi-times. This should be a warning, and we should fix the call site.

This pr is just trying to verify if this way is viable

Comment (k82cn) [reply pkg/scheduler/api/node_info.go:232]: > This should be a warning, and we should fix the call site.

If we return nil, how to know the issue there?

**Comment by hzxuzhonghu** [comment]
The benchmark result shows a 30% improve

before:

```
BenchmarkSnapShot-4   	      37	 289957413 ns/op
```

After 
```
BenchmarkSnapShot-4   	      54	 197747549 ns/op

```

**Pull request closed without merging**